File: app/src/main/java/ekzeget/ru/ekzeget/util/AppVersionCode.java
package ekzeget.ru.ekzeget.util;

import android.content.Context;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;

import ekzeget.ru.ekzeget.preferences.PDefaultValue;

public class AppVersionCode {
    public static int getApkVersionCode(Context context) {
        try {
            PackageInfo packageInfo = context.getPackageManager().getPackageInfo(context.getPackageName(), 0);
            return packageInfo.versionCode;
        } catch (PackageManager.NameNotFoundException ignored) {
        }
        return PDefaultValue.VERSION_CODE;
    }
}

File: app-kotlin/src/main/java/ru/ekzeget/utils/AppVersionCode.kt
package ru.ekzeget.utils

import android.content.Context
import android.content.pm.PackageManager
import ru.ekzeget.preferences.PDefaultValue

object AppVersionCode {
    fun getApkVersionCode(context: Context): Int {
        try {
            val packageInfo = context.packageManager.getPackageInfo(context.packageName, 0)
            return packageInfo.versionCode
        } catch (ignored: PackageManager.NameNotFoundException) {
        }

        return PDefaultValue.VERSION_CODE
    }
}